airflow/timetables/datasets.py

```diff
@@ -52,24 +55,20 @@ def deserialize(cls, data: dict[str, typing.Any]) -> Timetable:
         from airflow.serialization.serialized_objects import decode_timetable
 
         return cls(
-            timetable=decode_timetable(data["timetable"]), datasets=[Dataset(**d) for d in data["datasets"]]
+            timetable=decode_timetable(data["timetable"]),
+            datasets=[],  # don't need the datasets after deserialization
```

Comment:
Why don't we need them?

Reply:
they end up getting stored on a dataset_triggers attr of DAG. i can add a better comment.

Reply:
added